tools/generate-helpers/pg-table-bindings/list.go

```diff
@@ -60,3 +61,21 @@ func resourceMetadataFromString(resource string) permissions.ResourceMetadata {
 	}
 	panic("unknown resource: " + resource)
 }
+
+func clusterGetter(schema *walker.Schema) string {
+	for _, f := range schema.Fields {
+		if strings.Contains(f.Search.FieldName, "Cluster ID") {
+			return f.Getter("obj")
+		}
+	}
+	panic(schema.TypeName + "has no cluster. Is it directly scoped?")
+}
+
+func namespaceGetter(schema *walker.Schema) string {
+	for _, f := range schema.Fields {
+		if strings.Contains(f.Search.FieldName, "Namespace") {
+			return f.Getter("obj")
+		}
+	}
+	return ""
```

Comment:
I suppose the difference of behaviour between `clusterGetter` and `namespaceGetter` is because all directly scoped resource types are expected to have a `Cluster ID` field whereas only directly namespace scoped resource types are expected to have a `Namespace` field.
This means if `namespaceGetter` were to panic, the code generation would fail for directly cluster scoped resource types.

Reply:
Exactly, I'm thinking about having our own object in template with all stuff related to sac possibly lazy evaluated. Then we could have a method that returns if it's namespace scope. I don't want to put to many things here so I just take an approach where empty getter (`""`) means cluster scope but maybe I should have add explicit variable for that.